vicon2_driver/include/vicon2_driver/vicon2_driver.hpp

```diff
@@ -93,15 +97,19 @@ class ViconDriverNode : public mocap_control::ControlledLifecycleNode
   int droppedFrameCount_;
   int n_markers_;
   int n_unlabeled_markers_;
+  // unlabelled markers will get an unique number on detection
+  int unlabeled_counter_;
   std::string qos_history_policy_;
   std::string qos_reliability_policy_;
+  std::map<std::string, int> markers_list_;
+  
   int qos_depth_;
 
   void process_frame();
   void process_markers(const rclcpp::Time & frame_time, unsigned int vicon_frame_num);
   void marker_to_tf(
     mocap_msgs::msg::Marker marker,
-    int marker_num, const rclcpp::Time & frame_time);
+    int marker_num, const rclcpp::Time & frame_time, std::string marker_name);
```

Comment:
please, for non-scalar types use a const reference 

`int marker_num, const rclcpp::Time & frame_time, const std::string & marker_name)`

Reply:
Good practice. Changed